internal/services/iothub/iothub_endpoint_storage_container_resource.go

```diff
@@ -95,8 +97,11 @@ func resourceIotHubEndpointStorageContainer() *pluginsdk.Resource {
 			},
 
 			"encoding": {
-				Type:     pluginsdk.TypeString,
-				Optional: true,
+				Type:             pluginsdk.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          string(devices.EncodingAvro),
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Instead of suppressing case could we convert the value returned from the API into the correct case?

Reply:
validation function StringInSlice of encoding has `ignoreCase` set to true, removing suppress.CaseDifference + modifying the service returned value could break existing configuration files which have an input of different case.

Reply:
😪 well that is wrong and shouldn't have made it into the provider. could we add a comment to normalize this with pandora in 3.0, or more likely 4.0?

Reply:
Added the comment for normalizing this property and make it case-sensitive, please take a look